refactor-first-maven-plugin/src/main/java/org/hjug/mavenreport/ReportWriter.java

```diff
@@ -0,0 +1,43 @@
+package org.hjug.mavenreport;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.maven.project.MavenProject;
+
+@Slf4j
+public class ReportWriter {
+  public static void writeReportToDisk(
+      final MavenProject project,
+      final String filename,
+      final StringBuilder stringBuilder
+  ) {
+    final String reportOutputDirectory = project.getModel().getReporting().getOutputDirectory();
+    final File reportOutputDir = new File(reportOutputDirectory);
+
+    if(!reportOutputDir.exists()) {
+      reportOutputDir.mkdirs();
+    }
+
+    final String pathname = reportOutputDirectory + File.separator + filename;
+
+    final File reportFile = new File(pathname);
+
+    try {
+      reportFile.createNewFile();
+    } catch (IOException e) {
+      log.error("Failure creating chart script file", e);
+    }
+
+    try(BufferedWriter writer = Files.newBufferedWriter(reportFile.toPath(), UTF_8)) {' or 'try(BufferedWriter writer = Files.newBufferedWriter(reportFile.toPath(), Charset.defaultCharset())) {
```

Comment:
hmm I took this commit from the squirrel it seems to have broken some things. It didn't import the dependencies and such.

It's also a bit hard to read. Let me clean this up some

Reply:
OK, patched this up. I think it was trying to say I could a or b but the full comment got committed including the ' or ' stuff

`try(BufferedWriter writer = Files.newBufferedWriter(reportFile.toPath(), UTF_8)) {' or 'try(BufferedWriter writer = Files.newBufferedWriter(reportFile.toPath(), Charset.defaultCharset())) {`

It should be fixed now. I'll bring up this issue concerning the suggested commit with our team